components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/logout/idpinit/processor/FederatedIdpInitLogoutProcessor.java

```diff
@@ -346,6 +369,10 @@ protected void validateIssuerClaim(JWTClaimsSet claimsSet) throws LogoutClientEx
      */
     protected void validateAudience(List<String> aud, IdentityProvider idp) throws LogoutClientException {
 
+        // Validate audience is skipped for the resident IDP.
+        if (RESIDENT_IDP_RESERVED_NAME.equals(idp.getIdentityProviderName())) {
```

Comment:
Let's add the audience validation for the resident idp tokens as well. Need to check the complexity of doing this.